snowflake/src/main/java/org/apache/iceberg/snowflake/SnowflakeCatalog.java

```diff
@@ -114,9 +121,13 @@ public void initialize(String name, Map<String, String> properties) {
           cnfe);
     }
 
+    String uniqueId = UUID.randomUUID().toString().replace("-", "").substring(0, UNIQUE_ID_LENGTH);
```

Comment:
Rather than a unique identifier, I would suggest using the build info: `IcebergBuild.fullVersion()` and `IcebergBuild.gitCommitShortId()` so you know what client version and build is being used.

Reply:
Thanks for the review @danielcweeks ! I have added iceberg version and commit. The Unique identifier part is still required to correlate header with JDBC sessions in our telemetry. Also I have to move this PR to https://github.com/apache/iceberg/pull/7580 due to bad rebase.